### PR TITLE
rsc: skip next builtin module when apply loaders

### PR DESCRIPTION
**Pull request by huozhi**
x-ref: https://github.com/vercel/server-components-notes-demo/pull/17

Previously in #35975 we ignore handling node_modules as a workaround for 3rd party packages, but for next buildin components we should still handle them for processing client components

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| buildDuration | 20.2s | 19.3s | -908ms |
| buildDurationCached | 7.4s | 7.6s | ⚠️ +108ms |
| nodeModulesSize | 485 MB | 485 MB | ⚠️ +58 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.957 | 3.849 | -0.11 |
| / avg req/sec | 631.81 | 649.59 | +17.78 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.786 | 1.689 | -0.1 |
| /error-in-render avg req/sec | 1399.99 | 1479.95 | +79.96 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 28.3 kB | 28.3 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 72 kB | 72 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 192 B | 192 B | ✓ |
| amp-HASH.js gzip | 309 B | 309 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 3.04 kB | 3.04 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.74 kB | 5.74 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 392 B | 392 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16 kB | 16 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| _buildManifest.js gzip | 461 B | 461 B | ✓ |
| Overall change | 461 B | 461 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| index.html gzip | 530 B | 530 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 524 B | 524 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| buildDuration | 22.1s | 22.3s | ⚠️ +176ms |
| buildDurationCached | 7.5s | 7.6s | ⚠️ +142ms |
| nodeModulesSize | 485 MB | 485 MB | ⚠️ +58 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 4.012 | 4.044 | ⚠️ +0.03 |
| / avg req/sec | 623.12 | 618.24 | ⚠️ -4.88 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.82 | 1.811 | -0.01 |
| /error-in-render avg req/sec | 1373.83 | 1380.23 | +6.4 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.7 kB | 28.7 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 72.6 kB | 72.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 313 B | 313 B | ✓ |
| css-HASH.js gzip | 325 B | 325 B | ✓ |
| dynamic-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 921 B | 921 B | ✓ |
| image-HASH.js gzip | 5.78 kB | 5.78 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.44 kB | 2.44 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 393 B | 393 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16.1 kB | 16.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| _buildManifest.js gzip | 457 B | 457 B | ✓ |
| Overall change | 457 B | 457 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rsc/skip-next-builtin | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 526 B | 526 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: c1ac40ad403aceab3285ce1be358e3c3a432556d

File: packages/next/build/webpack/loaders/next-flight-server-loader.ts [hunk 1]
```diff
@@ -208,7 +212,7 @@ export default async function transformSource(
   }
 
   const isServerComponent = createServerComponentFilter(extensions)
-  const isClientComponent = createClientComponentFilter(extensions)
+  const isClientComponent = createClientComponentFilter()
```
Comment (shuding) [inline packages/next/build/webpack/loaders/next-flight-server-loader.ts:215]: What’s the reason behind this?

Comment (huozhi) [reply packages/next/build/webpack/loaders/next-flight-server-loader.ts:215]: client components can have any extension, not only the page extensions. you can have `a.client.mjs` but it is not a valid page extension yet

Comment (shuding) [reply packages/next/build/webpack/loaders/next-flight-server-loader.ts:215]: Yeah, so in that sense server components can have any extension too. We should do the same for both component types, but only apply the `pageExtension` check for page entries.

**Review by shuding — APPROVED** [review]

**Pull request merged**